### ISSUE TITLE
Incorrect error in Cargo.toml: `[lints.rust.unexpected_cfgs]` not valid in schema

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust-analyzer#22842]
I have this in our Cargo.toml:
```
[lints.rust.unexpected_cfgs]
level = "warn"
check-cfg = ['cfg(bootstrap)', 'cfg(target_has_reliable_f16_math)']
```
Cargo accepts this just fine, at least the nightly version. However, RA complains when I open that file:
> `{"rust":{"unexpected_cfgs":{"level":"warn","check-cfg":["cfg(bootstrap)","cfg(target_has_reliable_f16_math)"]}}}` is not valid under any of the schemas listed in the 'anyOf' keyword

(The warning also talks about `anyOf` which is quite confusing, it seems to leak RA implementation details.)

**rust-analyzer version**: rust-analyzer version: 0.3.2971-standalone [/home/r/.vscode-oss/extensions/rust-lang.rust-analyzer-0.3.2971-linux-x64/server/rust-analyzer]

**rustc version**: rustc 1.99.0-nightly (fcbe7917b 2026-07-15)

**editor or extension**: VSCodium

**relevant settings**:
```
{
    "rust-analyzer.rustc.source": "discover",
    "rust-analyzer.linkedProjects": [
        "Cargo.toml",
        "genmc-sys/Cargo.toml",
        "cargo-miri/Cargo.toml",
        "miri-script/Cargo.toml",
    ],
    "rust-analyzer.check.invocationStrategy": "once",
    "rust-analyzer.check.overrideCommand": [
        "./miri",
        "clippy",
        "--all-features",
        "--message-format=json",
    ],
    "rust-analyzer.cargo.features": "all",
    "rust-analyzer.cargo.extraEnv": {
        "MIRI_AUTO_OPS": "no",
        "MIRI_IN_RA": "1",
    },
    // Contrary to what the name suggests, this also affects proc macros.
    "rust-analyzer.cargo.buildScripts.invocationStrategy": "once",
    "rust-analyzer.cargo.buildScripts.overrideCommand": [
        "./miri",
        "check",
        "--all-features",
        "-Zunstable-options",
        "--compile-time-deps",
        "--message-format=json",
    ],
}
```

**repository link (if public, optional)**: https://github.com/rust-lang/miri/

**Comment by ChayimFriedman2** [comment]
I'm pretty sure r-a does not validate Cargo.toml and does not even read this key, and searching for this message in r-a's code reveals nothing; are you sure it's r-a and not some other extension you have?

**Comment by RalfJung** [comment]
Oh, it never occurred to me that other extensions might have support for specifically Cargo.toml... but I do have "Even Better TOML" installed and indeed that is the source of the error. Sorry for the noise!
